glean-core/src/upload/mod.rs

```diff
@@ -293,4 +360,78 @@ mod test {
         // Verify that after all requests are returned, none are left
         assert_eq!(upload_manager.get_upload_task(), PingUploadTask::Done);
     }
+
+    #[test]
+    fn test_processes_correctly_different_upload_responses() {
```

Comment:
nit: can you please break this test into smaller unit tests? see [this Kotlin code](https://github.com/mozilla/glean/blob/master/glean-core/android/src/test/java/mozilla/telemetry/glean/net/HttpURLConnectionUploaderTest.kt) for some related test coverage. You might also think to convert these tests to Rust.

Reply:
Yes. About converting the tests to Rust, do you see any specific test case that I haven't covered in my tests? I did based myself off of these and can't see any that made sense and I didn't copy over.